### PR TITLE
Initial node startup

### LINKED ISSUE #12: Support initializing a Node
```gherkin
Scenario: Initializing a Node
	When I initailize a Node
	Then I should be able to specify the "node mode"
	And I should be able to specify the name of the node
	And I should be able to specify the port
```

### PR DESCRIPTION
**Pull request by i1i1**
Closes #12

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -49,27 +118,218 @@ impl Builder {
...
+        self.port = Port(port);
+        self
+    }
+
+    pub fn at_directory(mut self, directory: impl Into<Directory>) -> Self {
+        self.directory = directory.into();
         self
     }
 
     /// It supposed to open node at the supplied location
```
Comment (ozgunozerk) [inline src/node.rs:130]: typo -> It *is* supposed to

File: src/node.rs [hunk 2]
```diff
@@ -49,27 +118,218 @@ impl Builder {
...
+            name,
+            port: Port(port),
+            validate,
+        } = self;
+        let mut args = vec![std::ffi::OsString::from("node-binary-place-holder")];
+
+        args.extend_from_slice(&["--state-pruning".into(), "archive".into()]);
+        match chain {
+            Chain::Gemini2a => args.extend_from_slice(&["--chain".into(), "gemini-2a".into()]),
+            Chain::Custom(chain) => match chain {},
```
Comment (ozgunozerk) [inline src/node.rs:159]: what does an empty `match chain {}` do? How is it useful in here?

File: src/node.rs [hunk 3]
```diff
@@ -49,27 +118,218 @@ impl Builder {
...
+        };
+        match &directory {
+            Directory::Tmp => args.extend_from_slice(&["--tmp".into()]),
+            Directory::Custom(path) => args.extend_from_slice(&["--base-path".into(), path.into()]),
+            Directory::Default => (), // TODO: add default path here
```
Comment (ozgunozerk) [inline src/node.rs:164]: was this TODO supposed to be resolved in this PR or not?

File: src/node.rs (hunk 2 above)
Comment (i1i1) [reply src/node.rs:159]: Let me open issue to support supplying custom chainspecs. For now there is stub which can't be constructed

File: src/node.rs (hunk 3 above)
Comment (i1i1) [reply src/node.rs]: Fixed it!

File: src/node.rs (hunk 2 above)
Comment (i1i1) [reply src/node.rs:166]: Here it is #18

**Comment by ozgunozerk** [comment]
looking good to me!

**Review by ozgunozerk — APPROVED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -7,12 +7,36 @@ edition = "2021"
 
 [dependencies]
 bytesize = "1.1"
+clap = { version = "3", features = ["derive"] }
+dirs = "4"
 either = "1.8"
 futures = "0.3"
-libp2p-core = "0.36"
+libp2p-core = "0.34"
```
Comment (achiurizo) [inline Cargo.toml:14]: curious, why the version bump down?

Comment (i1i1) [reply Cargo.toml:14]: Cause of the version conflict. Right now monorepo uses `libp2p-core = 0.34`

**Comment by i1i1** [comment]
Closed as there will be another take on this

**Pull request closed without merging**